server/aesdsocket.c

```diff
@@ -0,0 +1,346 @@
+    {
+        perror("set socket opt failed");
+        syslog(LOG_ERR,"set socket opt failed");
+        cleanup_on_exit();
+        return (EXIT_FAILURE);
+    }
+
+    /*binding the Socket port*/
+    memset(&hints, 0, sizeof(hints));
+    hints.ai_family = AF_INET;  // use IPv4 or IPv6, whichever
+    hints.ai_socktype = SOCK_STREAM;
+    hints.ai_flags = AI_PASSIVE;     // fill in my IP for me
+
+    if(getaddrinfo(NULL, "9000", &hints, &res)!= 0)
+    {
+        perror("socket get addr info failed");
+        syslog(LOG_ERR,"socket get addr info failed");
+        cleanup_on_exit();
+        return (EXIT_FAILURE);
+    }
+
+    if(bind(fd_soc_server, res->ai_addr, res->ai_addrlen) == -1)
+    {
+        perror("socket bind failed");
+        syslog(LOG_ERR,"socket bind failed");
+        cleanup_on_exit();
+        return (EXIT_FAILURE);
+    }
+
+    /*listen for the socket connection*/
+    if(listen(fd_soc_server, BACKLOG) == -1)
+    {
+        perror("socket listen failed");
+        syslog(LOG_ERR,"socket listen failed");
+        cleanup_on_exit();
+        exit(EXIT_FAILURE);
+    }
+    memset(&ip4addr, 0, sizeof(struct sockaddr_in));
+    memset(&addr_size, 0, sizeof(socklen_t));
+    /*super loop to poll on the data received from client*/
+    while (1)
+    {
+        if((fd_soc_client = accept(fd_soc_server, (struct sockaddr *)&ip4addr, &addr_size)) < 0)
+        {
+            syslog(LOG_ERR, "Failed to accept connection on socket");
+            cleanup_on_exit();
+            return (EXIT_FAILURE);
+        }
+
+        /*Converting IP address to string format*/
+        if(NULL == inet_ntop(AF_INET, &ip4addr.sin_addr, client_addr, sizeof(client_addr)))
+        {
+            syslog(LOG_ERR, "Failed to convert IPv4 address to string");
+            cleanup_on_exit();
+            return (EXIT_FAILURE);
+        }
+        syslog(LOG_INFO, "Accepted connection from %s", client_addr);           /*print the accepted client IPv4 Address*/
+
+        while(1)
+        {
+            while(1)
+            {
+                if((buffer_size = recv(fd_soc_client, buf, MAXDATASIZE, 0)) <= 0 )
+                {
+                    syslog(LOG_ERR, "Failed to receive data from socket");
+                    close(fd_soc_client);
+                    break;
+                }
+
+                if((ret_val = read_datasocket_strlocalbuf(fd_wr_file, buf, buffer_size)) == -1)
+                {
+                    close(fd_soc_client);               /*if read failed*/
+                    break;
+                }
+
+                if (ret_val > 0)
+                {
+                    recv_bytes += ret_val;
+                    break;
+                }
+            }
+
+            if (malloc_wr_buffer != NULL)
+            {
+                free(malloc_wr_buffer);
+            }
+
+            if((malloc_wr_buffer = (char *)malloc(sizeof(char) * recv_bytes)) == NULL)
```

Comment:
@Nileshkartik95 what assumption are you making here about the size of the process heap and the size of all packets sent?